### PR TITLE
chore(connlib): Add external ID to FFI, return `fd` in `on_set_interface_config`

### PR DESCRIPTION
**Pull request by jamilbk**
(Supersedes #1944)

* Fixes https://github.com/firezone/product/issues/649
* Passes `dns_fallback_strategy` over FFI (these are hardcoded for now)
* Incorporates @conectado 's #1944 and cleans up a few places `fd` was still passed

Draft for now until I can test it more tomorrow

### CODE REVIEW

**Comment by jamilbk** [comment]
@conectado Feel free to continue off this one. I didn't keep them separate since there'd be lots of conflicts around all the `connect()` flow.

File: rust/connlib/libs/common/src/session.rs [hunk 1]
```diff
@@ -9,16 +9,15 @@ use std::{
     fmt::{Debug, Display},
     marker::PhantomData,
     net::{Ipv4Addr, Ipv6Addr},
+    os::fd::RawFd,
```
Comment (jamilbk) [inline rust/connlib/libs/common/src/session.rs]: Unfortunately this fails for `target_family = "windows"`

File: rust/connlib/libs/client/src/messages.rs [hunk 2]
```diff
@@ -190,7 +191,8 @@ mod test {
                 "interface": {
                     "ipv4": "100.72.112.111",
                     "ipv6": "fd00:2021:1111::13:efb9",
-                    "upstream_dns": []
+                    "upstream_dns": [],
+                    "dns_fallback_strategy": "upstream_resolver"
```
Comment (conectado) [inline rust/connlib/libs/client/src/messages.rs]: @jamilbk I think adding this to the protocol should be made in another PR where we also add handling for it.

Comment (jamilbk) [reply rust/connlib/libs/client/src/messages.rs]: Hmm, yeah I see that it's a ball of yarn now. I'll revert this dns stuff.

Comment (jamilbk) [reply rust/connlib/libs/client/src/messages.rs]: Ah, you've done that. Thanks!

File: rust/connlib/libs/common/src/session.rs (hunk 1 above)
Comment (jamilbk) [reply rust/connlib/libs/common/src/session.rs]: @conectado Added a `DeviceRef` struct that wraps either `std::os::fd::RawFd` or `std::os::windows::io::RawHandle` depending on the os family. Can work on the remaining fixes this morning to get the builds to pass.

**Comment by jamilbk** [comment]
CI is green! Merging... 🤠
